src/enochecker/checkerservice.py

```diff
@@ -268,62 +100,66 @@ def serve_checker() -> Response:
                 exc_info=ex,
             )
-            return jsonify(
-                {
-                    "result": Result.INTERNAL_ERROR.name,
-                    "message": "Critical checker error occurred",
-                    "traceback": exception_to_string(ex),
-                }
+            result_message = CheckerResultMessage(
+                result=CheckerTaskResult.CHECKER_TASK_RESULT_INTERNAL_ERROR,
+                message=f"Critical checker error occured\n{exception_to_string(ex)}",
+            )
+            res_json = jsons.dumps(
+                result_message,
+                use_enum_name=False,
+                key_transformer=jsons.KEY_TRANSFORMER_CAMELCASE,
+                strict=True,
             )
+            return Response(res_json, mimetype="application/json")
 
-    def service_info() -> Dict[str, Union[str, int]]:
+    def service_info() -> CheckerInfoMessage:
         """
         Return technical information about this service.
 
-        Includes the name and desired number of flags, havoc and noise per round.
+        Includes the name and supported variants of flags, havoc and noise.
 
-        :return: dictionary with information
+        :return: CheckerInfoMessage filled with the values from the checker
         """
         try:
             service_name: str = getattr(
                 checker_cls, "service_name", checker_cls.__name__.split("Checker")[0]
             )
 
-            info_dict = {
-                "serviceName": service_name,
-                "flagCount": checker_cls.flag_count,
-                "havocCount": checker_cls.havoc_count,
-                "noiseCount": checker_cls.noise_count,
-            }
-
-            assert isinstance(info_dict["serviceName"], str)
-            assert isinstance(info_dict["flagCount"], int)
-            assert isinstance(info_dict["havocCount"], int)
-            assert isinstance(info_dict["noiseCount"], int)
-
-            return info_dict  # type: ignore # (mypy would infer Dict[str, object] instead)
+            return CheckerInfoMessage(
+                service_name=service_name,
+                flag_variants=checker_cls.flag_variants,
+                noise_variants=checker_cls.noise_variants,
+                havoc_variants=checker_cls.havoc_variants,
+            )
 
         except Exception:
             logger.error(
                 """Service info not specified!
-add service_name, flag_count, havoc_count and noise_count as static fields to your CHECKER
+add service_name, flag_variants, havoc_variants and noise_variants as static fields to your CHECKER
 
 Example:
 class ExampleChecker(BaseChecker):
-    flag_count  = 1
-    noise_count = 1
-    havoc_count = 1
+    flag_variants  = 1
+    noise_variants = 1
+    havoc_variants = 1
 """
             )
             raise AttributeError("REQUIRED SERVICE INFO FIELDS NOT SPECIFIED!")
 
-    def get_service_info() -> str:
+    def get_service_info() -> Response:
```

Comment:
This should either still return the `res_json` `str`, or the documentation should be changed.

Reply:
that is still the `res_json`, just wrapped into a Flask response. Does that really need clarification in the docstring?

Reply:
It's documented as JSON representation, while it returns a Flask response instead. Not good craftmanship, but since nobody will use this function outside, I don't care too much.